embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/internal/logs/LogSinkImpltest.kt

```diff
@@ -0,0 +1,71 @@
+package io.embrace.android.embracesdk.internal.logs
+
+import io.embrace.android.embracesdk.fixtures.testLog
+import io.mockk.every
+import io.mockk.mockk
+import io.opentelemetry.api.common.AttributeKey
+import io.opentelemetry.api.common.Attributes
+import io.opentelemetry.api.logs.Severity
+import io.opentelemetry.sdk.common.CompletableResultCode
+import io.opentelemetry.sdk.logs.data.Body
+import io.opentelemetry.sdk.logs.data.LogRecordData
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertSame
+import org.junit.Before
+import org.junit.Test
+
+internal class LogSinkImpltest {
+    private lateinit var logSink: LogSink
+
+    @Before
+    fun setup() {
+        logSink = LogSinkImpl()
+    }
+
+    @Test
+    fun `verify default state`() {
+        assertEquals(0, logSink.completedLogs().size)
+        assertEquals(0, logSink.flushLogs().size)
+        assertEquals(CompletableResultCode.ofSuccess(), logSink.storeLogs(listOf()))
+    }
+
+    @Test
+    fun `storing logs adds to stored logs`() {
+        val resultCode = logSink.storeLogs(listOf(getLogRecordData()))
+        assertEquals(CompletableResultCode.ofSuccess(), resultCode)
+        assertEquals(1, logSink.completedLogs().size)
+        assertEquals(EmbraceLogRecordData(logRecordData = getLogRecordData()), logSink.completedLogs().first())
+    }
+
+    @Test
+    fun `flushing clears stored logs`() {
+        logSink.storeLogs(listOf(mockk(relaxed = true), mockk(relaxed = true)))
+        val snapshot = logSink.completedLogs()
+        assertEquals(2, snapshot.size)
+
+        val flushedLogs = logSink.flushLogs()
+        assertEquals(2, flushedLogs.size)
+        repeat(2) {
+            assertSame(snapshot[it], flushedLogs[it])
+        }
+        assertEquals(0, logSink.completedLogs().size)
+    }
+
+    private fun getLogRecordData(): LogRecordData {
```

Comment:
Either in this PR or another one, can you create a `FakeLogRecordData` that based on the interface so you can use it instead of creating mock? We prefer this so it's easier to work with test data when we use real objects that we can instantiate and won't have to worry about the mocks being always right.

Reply:
`FakeLogRecordData` created.